blocks/main_loop.py

```diff
@@ -233,8 +234,20 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                callbackname = [key for key, value
+                                in TrainingExtension.__dict__.items()
+                                if getattr(value, '_is_callback', False)]
+                if other not in callbackname:
+                    if isinstance(other, str):
```

Comment:
This is bad practice, because `str` doesn't include unicode strings in Python 3, while `basestring` doesn't exist anymore in Python 3. If you really need to check for a string, you need to use `six.string_types`. In this case however, I don't think you need to check specifically for a string. If it's not a string it will just say `"<SomeObject> is not a valid callback"`, which is perfectly fine.